### PR TITLE
Fix windows service not working with local service

### PR DESCRIPTION
**Pull request by stiliev**
Fix windows service not working with local service, because of a permission issue to register an event source.

### CODE REVIEW

File: src/Agent.Listener/Configuration/NativeWindowsServiceHelper.cs [hunk 1]
```diff
@@ -389,6 +389,12 @@ public void InstallService(string serviceName, string serviceDisplayName, string
                     throw new InvalidOperationException(StringUtil.Loc("OperationFailed", nameof(CreateService), GetLastError()));
                 }
 
+                //invoke the service with special argument, that tells it to register an event log trace source (need to run as an admin)
+                using (var processInvoker = HostContext.CreateService<IProcessInvoker>())
+                {
+                    processInvoker.ExecuteAsync(string.Empty, agentServiceExecutable, $"init \"{serviceDisplayName}\"", null, default(System.Threading.CancellationToken)).GetAwaiter().GetResult();
```
Comment (TingluoHuang) [inline src/Agent.Listener/Configuration/NativeWindowsServiceHelper.cs]: you don't have to pass servicedisplayname, the service.exe control what name to use.

File: src/Agent.Service/Windows/Program.cs [hunk 2]
```diff
@@ -14,6 +10,12 @@ static class Program
         /// </summary>
         static void Main(String[] args)
         {
+            if (args != null && args.Length == 2 && args[0].Equals("init", StringComparison.InvariantCultureIgnoreCase))
+            {
+                string source = "VstsAgentService";
```
Comment (ericsciple) [inline src/Agent.Service/Windows/Program.cs]: It looks like this should share a const with the same string below.

Comment (ericsciple) [inline src/Agent.Service/Windows/Program.cs]: Does this always match the service name? What if multiple agent Windows services are installed on the same box? Or is support for that that not implemented yet?

File: src/Agent.Listener/Configuration/NativeWindowsServiceHelper.cs [hunk 3]
```diff
@@ -389,6 +389,12 @@ public void InstallService(string serviceName, string serviceDisplayName, string
                     throw new InvalidOperationException(StringUtil.Loc("OperationFailed", nameof(CreateService), GetLastError()));
                 }
 
+                //invoke the service with special argument, that tells it to register an event log trace source (need to run as an admin)
+                using (var processInvoker = HostContext.CreateService<IProcessInvoker>())
```
Comment (ericsciple) [inline src/Agent.Listener/Configuration/NativeWindowsServiceHelper.cs]: This is fine for now. Per offline conversation, it sounds like pinvoke isn't a clean option. I wonder if powershell or cmd.exe would be easier.

File: src/Agent.Service/Windows/Program.cs (hunk 2 above)
Comment (stiliev) [inline src/Agent.Service/Windows/Program.cs]: all services use the same trace source name for now

Comment (stiliev) [inline src/Agent.Service/Windows/Program.cs]: adding a const for "VstsAgentService"

**Pull request merged**